### DISCUSSION TITLE
MongoDB Schema Validation

### DISCUSSION THREAD

**Discussion by piercsi** [BeanieODM/beanie#632] [General]
Has there been any thought about taking a Document and creating (or updating) the corresponding schema validation in the MongoDB collection (https://www.mongodb.com/docs/manual/core/schema-validation/)?

**Comment by roman-right** [comment]
Hi! I thought about this, yes. I need to experiment on how well the schema for custom field types can be generated. If you have any ideas about the implementation in mind, I'd be happy to collaborate

**Comment by piercsi** [comment]
Yes I have some ideas, I have something that does this already.

I need to get a bit more familiar with beanie as I haven't used it yet, and will probably wait for the pydantic v2 version.

In the meantime I will work through some initial thoughts and get back in touch.
